File: test/jskills/trueskill/FactorGraphTrueSkillCalculatorTests.java
package jskills.trueskill;

import org.testng.annotations.Test;

public class FactorGraphTrueSkillCalculatorTests {

    @Test
    public void FullFactorGraphCalculatorTests() {
        FactorGraphTrueSkillCalculator calculator = new FactorGraphTrueSkillCalculator();

        // We can test all classes 
        TrueSkillCalculatorTests.TestAllTwoPlayerScenarios(calculator);
        TrueSkillCalculatorTests.TestAllTwoTeamScenarios(calculator);
        TrueSkillCalculatorTests.TestAllMultipleTeamScenarios(calculator);

        TrueSkillCalculatorTests.TestPartialPlayScenarios(calculator);
    }
}

File: src/test/scala/jskills/trueskill/FactorGraphTrueSkillCalculatorTests.scala
package jskills.trueskill

import org.junit.Before
import org.junit.Test

class FactorGraphTrueSkillCalculatorTests {
  var calculator: FactorGraphTrueSkillCalculator = null

  @Before def setup() {
    calculator = new FactorGraphTrueSkillCalculator()
  }

  @Test def TestAllTwoTeamScenarios() {
    TrueSkillCalculatorTests.TestAllTwoTeamScenarios(calculator)
  }

  @Test def TestAllTwoPlayerScenarios() {
    TrueSkillCalculatorTests.TestAllTwoPlayerScenarios(calculator)
  }

  @Test def TestAllMultipleTeamScenarios() {
    TrueSkillCalculatorTests.TestAllMultipleTeamScenarios(calculator)
  }

  @Test def TestPartialPlayScenarios() {
    TrueSkillCalculatorTests.TestPartialPlayScenarios(calculator)
  }
}